services/accumulo-service/src/main/java/datawave/microservice/audit/AuditClient.java

```diff
@@ -0,0 +1,196 @@
+package datawave.microservice.audit;
+
+import com.google.common.base.Preconditions;
+import datawave.marking.SecurityMarking;
+import datawave.microservice.authorization.jwt.JWTRestTemplate;
+import datawave.microservice.authorization.user.ProxiedUserDetails;
+import datawave.security.authorization.DatawaveUser;
+import datawave.security.util.ProxiedEntityUtils;
+import datawave.webservice.common.audit.AuditParameters;
+import datawave.webservice.common.audit.Auditor;
+import datawave.webservice.common.audit.Auditor.AuditType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Collection;
+
+/**
+ * Simple rest client for submitting requests to the audit service
+ *
+ * @see Request
+ * @see AuditServiceProvider
+ */
+@Service
+@ConditionalOnProperty(name = "audit.enabled", havingValue = "true")
+public class AuditClient {
+    
+    private static final String DEFAULT_REQUEST_PATH = "/v1/audit";
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    @Autowired
+    private AuditServiceProvider serviceProvider;
+    
+    private final JWTRestTemplate jwtRestTemplate;
+    
+    public AuditClient(RestTemplateBuilder builder) {
+        this.jwtRestTemplate = builder.build(JWTRestTemplate.class);
+    }
+    
+    public void submit(Request request) {
+        submit(request, DEFAULT_REQUEST_PATH);
+    }
+    
+    public void submit(Request request, String requestPath) {
+        
+        Preconditions.checkNotNull(request, "request cannot be null");
+        Preconditions.checkNotNull(requestPath, "requestPath cannot be null");
+        
+        if (log.isDebugEnabled()) {
```

Comment:
Same for all the log statements in this class.

Reply:
But slf4j doesn't prevent the first argument (string literal) from getting thrown onto the heap, right?

Reply:
slf4j won't convert the given parameter to a string unless the log message is going to get sent (e.g., the log level is debug in this case). If you do:
```
log.debug("Received audit request: {}", request.getAuditParameters().toString());
```
then there's no benefit, but if you do:
```
log.debug("Received audit request: {}", request.getAuditParameters());
```
then unless `getAuditParameters()` is expensive somehow, this will be just as effective and more readable than wrapping in `if (log.isDebugEnabled())`.